app/controllers/home_controller.rb

```diff
@@ -1,5 +1,21 @@
 class HomeController < ApplicationController
 
+  before_action :redirect_to_last_guide
+
   def index
   end
+
+  private
+
+  def redirect_to_last_guide
+    redirect_to current_user.last_guide, notice: t(:welcome_back_after_redirection) if visitor_recurrent?
+  end
+
+  def visitor_recurrent?
+    current_user? && current_user.last_guide.present? && visitor_comes_from_internet?
+  end
+
+  def visitor_comes_from_internet?
+    !%w(mumuki localhost).any? { |host| Addressable::URI.parse(request.referer).host.include? host } rescue true
```

Comment:
This deserves an auxiliary method like `request_host_include?`, so this would be 
`! request_host_include? %w(mumuki localhost)`